apis/network/v1alpha3/resourcerecordset_types.go

```diff
@@ -0,0 +1,137 @@
+	DNSRecordTypeCNAME DNSRecordType = "CNAME"
+
+	// DNSRecordTypeMX represents DNS MX record type.
+	DNSRecordTypeMX DNSRecordType = "MX"
+
+	// DNSRecordTypeNAPTR represents DNS NAPTR record type.
+	DNSRecordTypeNAPTR DNSRecordType = "NAPTR"
+
+	// DNSRecordTypePTR represents DNS PTR record type.
+	DNSRecordTypePTR DNSRecordType = "PTR"
+
+	// DNSRecordTypeSRV represents DNS SRV record type.
+	DNSRecordTypeSRV DNSRecordType = "SRV"
+
+	// DNSRecordTypeSPF represents DNS SPF record type.
+	DNSRecordTypeSPF DNSRecordType = "SPF"
+
+	// DNSRecordTypeAAAA represents DNS AAAA record type.
+	DNSRecordTypeAAAA DNSRecordType = "AAAA"
+
+	// DNSRecordTypeCAA represents DNS CAA record type.
+	DNSRecordTypeCAA DNSRecordType = "CAA"
+)
+
+// ChangeAction defines the valid actions that can be performed on a ResourceRecordSet.
+type ChangeAction string
+
+const (
+	// ChangeActionCreate represents a Resource Record CREATE operation.
+	ChangeActionCreate ChangeAction = "CREATE"
+
+	// ChangeActionDelete represents a Resource Record DELETE operation.
+	ChangeActionDelete ChangeAction = "DELETE"
+
+	// ChangeActionUpsert represents a Resource Record UPSERT operation.
+	ChangeActionUpsert ChangeAction = "UPSERT"
+)
+
+// +kubebuilder:object:root=true
+
+// ResourceRecordSet is a managed resource that represents an AWS Route53 Resource Record.
+// +kubebuilder:printcolumn:name="TYPE",type="string",JSONPath=".spec.forProvider.type"
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+type ResourceRecordSet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ResourceRecordSetSpec   `json:"spec"`
+	Status ResourceRecordSetStatus `json:"status,omitempty"`
+}
+
+// ResourceRecordSetSpec defines the desired state of an AWS Route53 Resource Record.
+type ResourceRecordSetSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  ResourceRecordSetParameters `json:"forProvider"`
+}
+
+// ResourceRecordSetStatus represents the observed state of a ResourceRecordSet.
+type ResourceRecordSetStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+}
+
+// ResourceRecord holds the DNS value to be used for the record.
+type ResourceRecord struct {
+	// The current or new DNS record value, not to exceed 4,000 characters. In the
+	// case of a DELETE action, if the current value does not match the actual value,
+	// an error is returned.
+	Value *string `json:"Value"`
+}
+
+// ResourceRecordSetParameters define the desired state of an AWS Route53 Resource Record.
+type ResourceRecordSetParameters struct {
```

Comment:
Some fields are missing such as `Weight`. Also I'd suggest looking at [types section here](https://github.com/crossplane/crossplane/blob/master/design/one-pager-managed-resource-api-design.md#pointer-types-and-markers) to see how their types and jsontags should look like.

Reply:
All of the fields that are available with AWS SDK's Resource Record Set has been added.